### PR TITLE
Add conda-recipes for DIRACOS2

### PR DESCRIPTION
**Pull request by chrisburr**
This PR adds recipes for conda-packages which we can't include in conda-forge. I've already built these and uploaded them to the [diracgrid](https://anaconda.org/diracgrid/) channels on anaconda.org where we have ~5GB of storage for free. See the README for more details.

Requires (already merged now):
* https://github.com/DIRACGrid/tornado_m2crypto/pull/1
* https://github.com/DIRACGrid/tornado_m2crypto/pull/2
* https://github.com/DIRACGrid/tornado/pull/1

### CODE REVIEW

**Comment by chrisburr** [comment]
@andresailer Apparently I don't have permission to assign you or request a review, but this is the first DIRACOS2-related PR for you take a look at.

In case you're not watching this repository, cc @chaen @fstagni @petricm

File: conda-recipes/README.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+
+If it is absolutely essential for DIRAC to have a custom build of a package, a new directory can be added here. An reference for the `meta.yaml` syntax can be found [here](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html) and the [conda-forge](https://conda-forge.org/feedstocks/) is a good source of examples. You can then test building packages by running:
+
+```bash
+conda build -c diracgrid -c conda-forge -m conda_build_config.yaml my-package/
+```
+
+The `extra-packages/conda_build_config.yaml` file is a "[variant config file](https://conda.io/projects/conda-build/en/latest/resources/variants.html#creating-conda-build-variant-config-files)" which is used to constrain the versions of dependencies. Currently this is only used to set the Python interpreter version. If shared library dependencies are added, new values should be added based on the main [conda-forge configuration](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
```
Comment (andresailer) [inline conda-recipes/README.md]:
```suggestion
The `conda-recipes/conda_build_config.yaml` file is a "[variant config file](https://conda.io/projects/conda-build/en/latest/resources/variants.html#creating-conda-build-variant-config-files)" which is used to constrain the versions of dependencies. Currently this is only used to set the Python interpreter version. If shared library dependencies are added, new values should be added based on the main [conda-forge configuration](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml).
```
?

File: conda-recipes/README.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
+# DIRACGrid's Conda recipes
+
+This directory contains the recipes that are used to generate the packages on the [`diracgrid`](https://anaconda.org/diracgrid/) conda channel.
+This should only be used in situations where it is impractical to add packages to [conda-forge](https://conda-forge.org/) directly.
+New packages can be added to conda-forge using [`staged-recipes`](https://github.com/conda-forge/staged-recipes/) and existing packages can be modified by making a PR to the corresponding [`feedstock`](https://conda-forge.org/feedstocks/).
+
+If it is absolutely essential for DIRAC to have a custom build of a package, a new directory can be added here. An reference for the `meta.yaml` syntax can be found [here](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html) and the [conda-forge](https://conda-forge.org/feedstocks/) is a good source of examples. You can then test building packages by running:
```
Comment (andresailer) [inline conda-recipes/README.md]:
```suggestion
If it is absolutely essential for DIRAC to have a custom build of a package, a new directory can be added here. A reference for the `meta.yaml` syntax can be found [here](https://conda.io/projects/conda-build/en/latest/resources/define-metadata.html) and the [conda-forge](https://conda-forge.org/feedstocks/) is a good source of examples. You can then test building packages by running:
```

**Review by andresailer — CHANGES_REQUESTED** [review]
In the PR comment you write "I've already built these and uploaded them". 

Please clarify in the README: if a new package or version of an existing is added to these recipes, are they then automatically uploaded to the diracgrid channel, or does one have to do something manually?

**Comment by chrisburr** [comment]
Thanks for taking a look @andresailer, I've made the changes.

For the build and upload of packages, I don't think it's worth the hassle of automating for now but if we start uploading packages frequently we can revaluate.

**Review by andresailer — APPROVED** [review]

**Comment by andresailer** [comment]
Yes, I agree.

**Pull request merged**